## 1 데이터 확인 및 전처리

In [7]:
import pandas as pd
import numpy as np
data = pd.read_csv('https://raw.githubusercontent.com/ADPclass/ADP_book_ver01/main/data/27_problem1.csv')
data

Time        V1        V2        V3        V4        V5        V6  \
0         60.0  1.069141  0.043786  0.309867  1.316694 -0.462394 -0.857888   
1        406.0 -2.312227  1.951992 -1.609851  3.997906 -0.522188 -1.426545   
2        426.0 -1.932055  1.724618  1.288229 -0.396998 -0.140202  2.470521   
3        472.0 -3.043541 -3.157307  1.088463  2.288644  1.359805 -1.064823   
4        674.0 -0.719065  0.638954  0.686335 -0.023240  2.317890  3.871978   
...        ...       ...       ...       ...       ...       ...       ...   
1188  172266.0  1.880157 -0.485374  0.095951  0.473297 -1.277631 -1.163752   
1189  172288.0  2.100535 -0.014620 -1.602113  0.184332  0.383228 -0.768377   
1190  172329.0  2.078542 -1.119785 -0.338306 -0.727997 -1.330486 -0.669741   
1191  172539.0 -5.823560  0.333130 -0.732746  0.770116 -3.785871  2.124706   
1192  172676.0  1.915474 -0.217019 -0.521731  1.274430 -0.124715  0.351640   

            V7        V8        V9       V10       V11       V12       V13  \
0     0.207904 -0.191270  0.056398  0.041567 -0.507358 -0.108666 -0.473257   
1    -2.537387  1.391657 -2.770089 -2.772272  3.202033 -2.899907 -0.595222   
2    -1.865298 -5.186855  0.794764  0.155379  0.034517  0.906738 -0.027172   
3     0.325574 -0.067794 -0.270953 -0.838587 -0.414575 -0.503141  0.676502   
4    -0.312797  1.089780 -0.596925 -0.006142 -0.242883 -0.384491  0.109079   
...        ...       ...       ...       ...       ...       ...       ...   
1188 -0.530703 -0.188080  1.340928 -0.242598 -0.032176  1.189035  0.778034   
1189  0.207428 -0.299064  0.584681  0.003055 -0.965941  0.225244  0.274983   
1190 -1.068741 -0.125540  0.191309  0.706048 -0.754812 -0.291895  0.614961   
1191 -2.272833  2.896201  0.552732  1.113104 -1.554701  1.183078  0.287108   
1192 -0.502276  0.170470  0.773668  0.280500  0.540609  1.202146  0.075329   

           V14       V15       V16       V17  Amount  Class  
0     0.452046  1.005675  0.185441 -0.439976   89.40      0  
1    -4.289254  0.389724 -1.140747 -2.830056    0.00      1  
2    -0.404303 -0.473949  0.274034 -0.222340   16.19      0  
3    -1.692029  2.000635  0.666780  0.599717  529.00      1  
4     0.187565  1.823472  0.274346 -0.697730   22.00      0  
...        ...       ...       ...       ...     ...    ...  
1188 -0.327769  0.351492 -0.213423 -0.176871   35.00      0  
1189  0.428356  0.793141 -0.104875 -0.647107    4.49      0  
1190 -0.634456  0.267806  1.148287  0.127145   47.22      0  
1191 -0.023928 -1.800240 -1.032457  0.975180  150.00      0  
1192 -0.023488 -1.020959 -0.021296 -0.580013    5.90      0  

[1193 rows x 20 columns]

### 1.1 
EDA 데이터 탐색

### 1.2 
변수간 상관관계를 시각화하고 전처리가 필요함을 설명하라

### 2.1 
차원축소 방법 2가지 이상 비교하고 한가지 선택

### 2.2 
추천한 한 가지를 실제로 수행하고 선택한 이유 설명

### 3.1 
오버샘플링과 언더샘플링 장단점 비교 및 선택 구현

### 3.2 
구현 및 알고리즘 2가지 이상 비교, 성능 측정

### 3.3 
현재까지 전처리한 데이터를 통해 모델 수행 후 결과 분석

### 4.1 
이상탐지 모델 2가지 이상 기술, 장/단점 설명

### 4.2 
2번에서 만든 데이터로 한 가지 이상탐지 모델을 구현하고, 3번에서 만든 모델과 비교

### 4.3 
데이터분석과 관점에서 3번에서 만든 모델과 4번에서 만든 모델 설명

### 5
2년 전 제품 생산량이 100,000개, 1년 전 제품 생산량이 150,000개, 그 후 팩토리 기술의 상승으로 제품 생산량이 250,000개 되었을 때, 연평균 상승률의 대푯값을 구하시오. (반올림하여 소수점 아래 둘째자리까지 표기하시오.)

### 6
12건의 광고 시간 측정. 평균 15.5초, 분산 3.2초 일때 90% 신뢰구간은?

## 7
강의 상류와 하류의 생물 다양성 점수에 차이가 있는지 검정하시오. (단, 같은 강에서 상류와 하류는 서로 독립적이지 않으며, 종속적인 관계에 있다.)

### 7-1
7.1 귀무가설과 대립가설을 세우시오

### 7-2
7.2 검정 통계량과 유의 확률을 구하고, 연구가설 채택여부를 검정하시오.